### PR TITLE
Added Terminal Velocity

### PR DESCRIPTION
**Pull request by AnangoPrabhat**
- updated the problem list to show terminal velocity on the site
- added the editorial for terminal velocity
- added the cpp solution for terminal velocity

### CODE REVIEW

File: content/problems/bio2/index.tsx [hunk 1]
```diff
@@ -61,5 +61,15 @@ export const bio2Problems = {
...
+  } as ProblemInfo,
+  terminalvelocity: {
+    difficulty: "Easy",
+    display: "Terminal Velocity",
+    original: "https://olympiad.org.uk/papers/2019/final/velocity.pdf",
+    year: 2019,
+    tags: [],
+    editorialAuthors: ["Anango Prabhat"],
+    solutionAuthors: ["Anango Prabhat"],
+    component: <TerminalVelocity />
```
Comment (myst-6) [inline content/problems/bio2/index.tsx:73]: On line $2$ you need to add TerminalVelocity into the list of imports.

File: content/problems/bio2/index.tsx [hunk 2]
```diff
@@ -61,5 +61,15 @@ export const bio2Problems = {
     editorialAuthors: ["Boris Hall"],
     solutionAuthors: ["Boris Hall"],
     component: <Milk />
+  } as ProblemInfo,
+  terminalvelocity: {
+    difficulty: "Easy",
+    display: "Terminal Velocity",
+    original: "https://olympiad.org.uk/papers/2019/final/velocity.pdf",
+    year: 2019,
+    tags: [],
```
Comment (myst-6) [inline content/problems/bio2/index.tsx:70]: Please add some tags.

File: solutions/bio2/TerminalVelocity/index.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        Since we are given all of the inversions, for a particular pigeon $x$ we know exactly which other 
+        pigeons will come before it in the final permutation. These are exactly the pigeons $y$ with a smaller 
+        starting position $y<x$ such that $(x,y)$ is not an inversion, and the pigeons $y$ with a larger starting 
+        position $y>x$ such that $(x,y)$ is an inversion.
+      </SText>
+      <SText>
+        Therefore, for each $x$ we can count the number of pigeons which come before $x$ in the final ordering. 
+        Let's process the inversion pairs one by one. At the beginning, with no inversions processed, the final 
+        position is the same as the initial position. Then, when processing pair $(x,y)$, we swap $x$ and $y$ if $y\x$ 
+        to ensure that $x<y$. Now we know that one more pigeon comes before $x$ than we thought before (pigeon $y$),
```
Comment (myst-6) [inline solutions/bio2/TerminalVelocity/index.tsx:35]: You can't write `<` in the text. You should write `$x \lt y$` instead.

File: solutions/bio2/TerminalVelocity/index.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        Therefore, for each $x$ we can count the number of pigeons which come before $x$ in the final ordering. 
+        Let's process the inversion pairs one by one. At the beginning, with no inversions processed, the final 
+        position is the same as the initial position. Then, when processing pair $(x,y)$, we swap $x$ and $y$ if $y\x$ 
+        to ensure that $x<y$. Now we know that one more pigeon comes before $x$ than we thought before (pigeon $y$), 
+        and we know that one less pigeon comes before $y$ than we thought before (pigeon $x$), so we can update our counts. 
+        In this way we know for each pigeon $i$ how many pigeons $j$ come before $i$ in the final ordering, and thus we 
+        know the position of each initial pigeon in the final ordering.
+      </SText>
+
+      <STitle>Time Complexity</STitle>
```
Comment (myst-6) [inline solutions/bio2/TerminalVelocity/index.tsx:41]: I don't think this needs to be under a separate heading. Just put it at the end of the previous section.

File: solutions/bio2/TerminalVelocity/index.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      <SText>
+        $O(n^2)$
+      </SText>
+    </>
```
Comment (myst-6) [inline solutions/bio2/TerminalVelocity/index.tsx:45]: After the last `SText`, you need to add an `SCodeBlock` and provide the path to the code you're rendering.

**Review by myst-6 — CHANGES_REQUESTED** [review]

**Comment by AnangoPrabhat** [comment]
List of commits:
https://github.com/myst-6/ukoly/pull/18/commits/a6419518b6cc091cf1ff56974806f482e9650b57
https://github.com/myst-6/ukoly/pull/18/commits/e2769ebb479ddc80579f51e195953abc7d7e4fc3
https://github.com/myst-6/ukoly/pull/18/commits/7855f50220d6759e4b212de9e2d066b3064ff271
https://github.com/myst-6/ukoly/pull/18/commits/201bbc6cf47924364826b29d466ae4ff8b1d0b8e
https://github.com/myst-6/ukoly/pull/18/commits/0f730ac8128f01376131aa24f1e2e35bec2cc739
https://github.com/myst-6/ukoly/pull/18/commits/f93f9ec13b546fff36f7867ae7fa1a2af6abac27
https://github.com/myst-6/ukoly/pull/18/commits/32126546bf3c47c6acd21546e19c0d77eedea60c

**Comment by AnangoPrabhat** [comment]
new commit:
https://github.com/myst-6/ukoly/pull/18/commits/cefa66779524d02e6c817fcc7120dd25467bf06e

**Comment by AnangoPrabhat** [comment]
new commit
https://github.com/myst-6/ukoly/pull/18/commits/a6419518b6cc091cf1ff56974806f482e9650b57

**Comment by AnangoPrabhat** [comment]
new commit:
https://github.com/myst-6/ukoly/pull/18/commits/93e134933535e27df5342631433060d59033fea8

File: solutions/bio2/TerminalVelocity/index.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      </SText>
+      <SText>
+        Therefore, for each $x$ we can count the number of pigeons which come before $x$ in the final ordering. 
+        Let's process the inversion pairs one by one. At the beginning, with no inversions processed, the final 
+        position is the same as the initial position. Then, when processing pair $(x,y)$, we swap $x$ and $y$ if $y \lt x$ 
+        to ensure that $x \lt y$. Now we know that one more pigeon comes before $x$ than we thought before (pigeon $y$), 
+        and we know that one less pigeon comes before $y$ than we thought before (pigeon $x$), so we can update our counts. 
+        In this way we know for each pigeon $i$ how many pigeons $j$ come before $i$ in the final ordering, and thus we 
+        know the position of each initial pigeon in the final ordering.
+      Time Complexity: $O(n^2)$
```
Comment (myst-6) [inline solutions/bio2/TerminalVelocity/index.tsx:38]: Sorry for the nitpick but could you change this to $O(p^2)$ please 🥺

**Review by myst-6 — CHANGES_REQUESTED** [review]
everything is fine except one small thing

File: solutions/bio2/TerminalVelocity/index.tsx (hunk 6 above)
Comment (myst-6) [reply solutions/bio2/TerminalVelocity/index.tsx]: Actually, please use `$\mathcal{brace("O")}(p^2)$`. It makes the O more fancy.

**Comment by AnangoPrabhat** [comment]
https://github.com/myst-6/ukoly/pull/18/commits/7b57623d4c0b00bc4d3b4695b29be4550dac762c
new commit link

File: solutions/bio2/TerminalVelocity/index.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+      </SText>
+      <SText>
+        Therefore, for each $x$ we can count the number of pigeons which come before $x$ in the final ordering. 
+        Let's process the inversion pairs one by one. At the beginning, with no inversions processed, the final 
+        position is the same as the initial position. Then, when processing pair $(x,y)$, we swap $x$ and $y$ if $y \lt x$ 
+        to ensure that $x \lt y$. Now we know that one more pigeon comes before $x$ than we thought before (pigeon $y$), 
+        and we know that one less pigeon comes before $y$ than we thought before (pigeon $x$), so we can update our counts. 
+        In this way we know for each pigeon $i$ how many pigeons $j$ come before $i$ in the final ordering, and thus we 
+        know the position of each initial pigeon in the final ordering.
+      Time Complexity: $$\mathcal{brace("O")}(p^2)$$
```
Comment (myst-6) [inline solutions/bio2/TerminalVelocity/index.tsx:39]: use a single set of dollar signs instead of double!!!

**Review by myst-6 — CHANGES_REQUESTED** [review]
i hope this is the last change

**Comment by AnangoPrabhat** [comment]
https://github.com/myst-6/ukoly/pull/18/commits/7835ec455d49bf66c3ef5708ccdbda2e86644e2d

**Review by myst-6 — APPROVED** [review]

**Pull request merged**